Code/Editor/EditorFramework/Assets/Implementation/AssetUpdates.cpp

```diff
@@ -632,6 +632,11 @@ void ezAssetCurator::InvalidateAssetTransformState(const ezUuid& assetGuid)
   ezAssetInfo* pAssetInfo = nullptr;
   if (m_KnownAssets.TryGetValue(assetGuid, pAssetInfo))
   {
+    if (pAssetInfo->m_bInvalidating)
```

Comment:
@Sanakan8472 I ran into a recursive crash once, after setting a particular asset dependency. I wasn't able to really fix it, but this part remained from my attempt.

Reply:
This could only happen if you have circular dependencies / references between assets. While this will prevent the stackoverflow it will most likely crash at a different location afterwards as the asset system is not designed to handle cyclic dependencies. It would be useful to figure out how the circular dependency came to be and what the use case is.
We absolutely can't have circles in m_InverseDependency. If there are circles in m_InverseReferences we could probably work around it but it would require a lot of special case handling but I doubt there are any intentional use cases for this.

Reply:
True, as I said, it continue to crash in other places. I think I got a cyclic dependency by having a Surface which referenced a prefab, which referenced a mesh, which referenced a material, which referenced the same surface....

The use case is real, it makes sense to have that kind of dependency chain, however, as we already brainstormed about in the past, it would be useful to have one of these links not as a fully asset dependency, but rather something else (since e.g. changing the surface on a material currently also triggers a full asset reprocessing, although it doesn't have any transitive impact).

Should I remove this change again, or keep it in?

Reply:
Remove for now, I'll think about it a bit.